### PR TITLE
Add support for phi 3 models

### PR DESCRIPTION
**Pull request by dmchoiboi**
# Pull Request Summary

_What is this PR changing? Why is this change being made? Any caveats you'd like to highlight? Link any relevant documents, links, or screenshots here if applicable._

## Test Plan and Usage Guide

_How did you validate that your PR works correctly? How do you run or demo the code? Provide enough detail so a reviewer can reasonably reproduce the testing procedure. Paste example command line invocations if applicable._

### CODE REVIEW

**Review by seanshi-scale — APPROVED** [review]

File: model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py [hunk 1]
```diff
@@ -210,6 +210,12 @@
             "gemma-2b-instruct",
             "gemma-7b",
             "gemma-7b-instruct",
+            "phi-3-mini-4k-instruct",
```
Comment (yunfeng-scale) [inline model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py:213]: these models don't have # billion parameters written in name, can you add special casing in `recommended_hardware_config_map`?

Comment (yunfeng-scale) [reply model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py:213]: or provide # b parameters in `_infer_hardware`

Comment (dmchoiboi) [reply model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py:213]: added parameter count for these models

File: docs/model_zoo.md [hunk 2]
```diff
@@ -42,6 +42,8 @@ Scale hosts the following models in the LLM Engine Model Zoo:
 | `gemma-2b-instruct`      | ✅                       |                            | vllm                                       | 8192                                           |
 | `gemma-7b`               | ✅                       |                            | vllm                                       | 8192                                           |
 | `gemma-7b-instruct`      | ✅                       |                            | vllm                                       | 8192                                           |
+| `phi-3-mini-4k-instruct` | ✅                       |                            | vllm                                       | 4096                                           |
```
Comment (yunfeng-scale) [inline docs/model_zoo.md:45]: do we want to add all of the model variants here?

**Review by yunfeng-scale — APPROVED** [review]

File: docs/model_zoo.md (hunk 2 above)
Comment (dmchoiboi) [reply docs/model_zoo.md:45]: We've only been asked to host phi-3-mini-4k-instruct so far

**Pull request merged**